### PR TITLE
Enhance error message #2426

### PR DESCRIPTION
**Pull request by keiya01**
- I added warning log that output error message when `formState.isValid` is used in `onSubmit` mode.
- I added warning log that output error message when registered field name is flat array with `useFieldArray`.

issue: #2426

### CODE REVIEW

File: src/useForm.ts [hunk 1]
```diff
@@ -1130,6 +1146,17 @@ export function useForm<
...
       ? new Proxy<FormStateProxy<TFieldValues>>(formState, {
           get: (obj, prop: keyof FormStateProxy) => {
+            if (
+              process.env.NODE_ENV !== 'production' &&
+              prop === 'isValid' &&
+              isOnSubmit
+            ) {
+              // eslint-disable-next-line no-console
+              console.warn(
+                'formState.isValid is only applicable with onChange and onBlur mode. Please see https://react-hook-form.com/api#formState',
```
Comment (bluebill1049) [inline src/useForm.ts:1156]:
```suggestion
                'formState.isValid is applicable with onChange and onBlur mode. https://react-hook-form.com/api#formState',
```

File: src/useForm.ts [hunk 2]
```diff
@@ -757,6 +758,21 @@ export function useForm<
...
+      fieldArrayNamesRef.current.has(stringToPath(ref.name)[0] as string) &&
+      !RegExp(
+        `^${stringToPath(ref.name)[0] as string}[\\d+]\.\\w+`
+          .replace(/\[/g, '\\[')
+          .replace(/\]/g, '\\]'),
+      ).test(ref.name)
+    ) {
+      // eslint-disable-next-line no-console
+      return console.warn(
+        "Make sure name is in object shape: name=test[index].name as we don't support flat arrays. Please see https://react-hook-form.com/api#useFieldArray",
```
Comment (bluebill1049) [inline src/useForm.ts:772]:
```suggestion
        "Name should be in object shape: name="test[index].name". https://react-hook-form.com/api#useFieldArray",
```

**Review by bluebill1049 — COMMENTED** [review]
This is awesome, I will generate a beta version to test out 👍

**Comment by bluebill1049** [comment]
`6.2.1-beta.1` tested looks good.

**Review by bluebill1049 — APPROVED** [review]
LGTM 👍

**Comment by keiya01** [comment]
Thank you very much for testing and reviewing 🥳

**Pull request merged**